### PR TITLE
UIImageView+Networking: Simplifying our Tools!

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
What's up!

- We're extracting an A8C **UIImageView** extension that allows us to download assets, polishing it's API, and moving it over to WordPressUI.
- This extension does not rely upon any 3rd party dependency.
- We're nuking a legacy **UIImageView+Networking** extension, which did the exact same thing, but relied on AFNetworking **(boooo)**.

### Scenario: Epilogue User Cell
1. Perform a fresh install
2. Log into a dotcom account

Verify that, when you reach the Epilogue UI (list of blogs!) your gravatar shows up, and you look as pretty as usual.

### Scenario: Notifications
1. Log into a dotcom account
2. Open the Notifications tab
3. Verify that the gravatars show up correctly (left side!)
4. Open any Comment-Y Notification
5. Verify that the gravatars show up just fine.

### Scenario: Plans
1. Log into a dotcom account
2. Open `My Sites` > `Your Site` > `Plans`

Verify that each plan row gets an asset. Press on any of them, and verify the images load properly.

### Scenario: Jetpack Plugins
1. Log into a dotcom account that has a Jetpack Enabled Site
2. Open `My Sites` > `Your Jetpack Site` > `Plugins`

Verify the images load up correctly!

### Scenario: Sharing Extension
1. Log into a dotcom account
2. Open Safari
3. Launch the Share Extension

Verify that all of your sites get their pretty Blavatar loaded.

--

@bummytime since this one really affects code that was bundled in the Share Extension, may i bug you with this one?.

It's mostly... moving code, and doing tiny cleanups here and there. Thanks **A LOT** in advance!!!

P.s.: YES! this is part of the Login.framework project!! 😄

### CODE REVIEW

**Comment by jleandroperez** [comment]
Since @bummytime is in his Helpshift rotation.. may i bug you with this one @jklausa?

Thanks in advance!!

File: WordPressUI/WordPressUI/Extensions/UIImageView+Networking.swift [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+public extension UIImageView {
+
+    /// Downloads an image and updates the current UIImageView Instance.
+    ///
+    /// - Parameters:
+    ///     -   url: The target Image's URL.
+    ///     -   placeholderImage: Image to be displayed while the actual asset gets downloaded.
+    ///     -   pointSize: *Maximum* allowed size. if the actual asset exceeds this size, we'll shrink it down.
+    ///
+    public func downloadResizedImage(at url: URL?, placeholderImage: UIImage? = nil, pointSize: CGSize) {
```
Comment (jklausa) [inline WordPressUI/WordPressUI/Extensions/UIImageView+Networking.swift]: `at:` as a parameter name feels a bit awkward to me — I don't think I'd say "I'm downloading a picture at URL: foo.bar/xyz". 

"I'm downloading an image `from` URL" seems more natural to me. WDYT?

**Comment by jleandroperez** [comment]
Thanks **a lot** for the review Jan!!. Public API updated, loved your suggestion!!

**Comment by jleandroperez** [comment]
@jklausa i've noticed i've missed few image assignments (Theme Browser + Site Creation). As a result, images there weren't being rendered.

By default, if the success block is present, the image is not getting set automatically. (Same behavior in AFNetworking!).

Mind taking a real quick peek?. (Thank you!!!)

**Comment by jklausa** [comment]
There you go sir!

**Comment by jleandroperez** [comment]
Wooooo thank you!!

**Comment by jklausa** [comment]
oh hrm, I just noticed something else:

the share extension isn't displaying blavatar for one of my sites (Gryffindor P2), but this is also happening in the current internal, so doesn't seem to be caused by this:

simulator screen shot - iphone 8 plus - 2018-04-06 at 03 36 01
simulator screen shot - iphone 8 plus - 2018-04-06 at 03 35 46

**Comment by jleandroperez** [comment]
Oops, merged a split second before. I'll try to reproduce this on my end, thanks for the heads up!!

**Pull request merged**